controls/SV-257777.rb

```diff
@@ -13,14 +13,43 @@
   desc 'fix', 'Upgrade to a supported version of RHEL 9.'
   impact 0.7
   ref 'DPMS Target Red Hat Enterprise Linux 9'
-  tag check_id: 'C-61518r925316_chk'
   tag severity: 'high'
+  tag gtitle: 'SRG-OS-000480-GPOS-00227'
   tag gid: 'V-257777'
   tag rid: 'SV-257777r925318_rule'
   tag stig_id: 'RHEL-09-211010'
-  tag gtitle: 'SRG-OS-000480-GPOS-00227'
   tag fix_id: 'F-61442r925317_fix'
-  tag 'documentable'
   tag cci: ['CCI-000366']
   tag nist: ['CM-6 b']
+  tag 'host'
+  tag 'container'
+
+  release = os.release
+
+  # Note that versions 9.0 and 9.2 of RHEL9 are within the EUS window at
+  # time of writing.
+
+  # 9.1 is not a EUS-supported release and is no longer officially supported
+  # by Red Hat. The date given for the expiration for 9.1 is based on the
+  # RHEL9 Planning Guide diagram found on Red Hat's Life Cycle page:
+  # https://access.redhat.com/support/policy/updates/errata/#Life_Cycle_Dates
+
+  EOMS_DATE = {
+    /^9\.0/ => '31 May 2024',
+    /^9\.1/ => 'April 1, 2023',
+    /^9\.2/ => 'May 31, 2025',
+    /^9\.4/ => 'May 31, 2026'
+  }.find { |k, _v| k.match(release) }&.last
+
+  describe "The release \"#{release}\"" do
+    if EOMS_DATE.nil?
+      it 'is a supported release' do
```

Comment:
No. `The release is a supported release` is what the test will print in the output, and the test will fail because EOMS_DATE is a nil (ergo its not a release with a support window published). We're forcing a fail if EOMS_DATE is nil but doing so such that the user will understand what the problem was.